File: araok/src/main/java/ru/otus/dto/AgeLimitDto.java
package ru.otus.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import ru.otus.domain.AgeLimit;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AgeLimitDto {
    private Long id;

    private String description;

    private Long limit;

    public static AgeLimit toDomainObject(AgeLimitDto dto) {
        return AgeLimit.builder()
                .id(dto.id)
                .description(dto.description)
                .limit(dto.limit)
                .build();
    }

    public static AgeLimitDto toDto(AgeLimit ageLimit) {
        return AgeLimitDto.builder()
                .id(ageLimit.getId())
                .description(ageLimit.getDescription())
                .limit(ageLimit.getLimit())
                .build();
    }
}

File: araok-android/src/main/java/ru/otus/data/dto/AgeLimitDto.kt
package ru.araok.data.dto

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import ru.araok.entites.AgeLimit

@JsonClass(generateAdapter = true)
data class AgeLimitDto(
    @Json(name = "id") override val id: Int?,
    @Json(name = "description") override val description: String,
    @Json(name = "limit") override val limit: Long
): AgeLimit